src/Esprima/ErrorHandlerExtensions.cs

```diff
@@ -2,18 +2,9 @@
 {
     public static class ErrorHandlerExtensions
     {
-        public static ParserException CreateError(this IErrorHandler handler, int index, int line, int col, string description)
+        public static void ThrowError(this IErrorHandler handler, int index, int line, int column, string message)
         {
-            var msg = $"Line {line}': {description}";
-            var error = new ParserException(msg)
-            {
-                Index = index,
-                Column = col,
-                LineNumber = line,
-                Description = description,
-                Source = handler.Source
-            };
-            return error;
+            throw handler.CreateError(index, line, column, message);
```

Comment:
could someone choose not to create exception and return null, causing this to throw null?

Reply:
Good point. I see one of two ways to approach that: (a) be super paranoid and check against undefined behaviour or (b) leave it as _undefined behaviour_ (albeit undocumented 😞 unless we add doc comments). Then again, we don't check that whether the `handler` is null. :wink:

Reply:
@lahma said (on Gitter chat):

> the create error comment was mostly for whether I was unsure if one would use the API to suppress errors and returning null would be the only way (which wouldn't work)

Seems that [Esprima doesn't allow `null` to be returned](https://github.com/jquery/esprima/blob/c0dfa42da4da9fa242dac623278a082cb38edc20/src/error-handler.ts#L49):

```js
createError(index: number, line: number, col: number, description: string): Error
```

In other words, the return type is not `Error | null` and I'm assuming they compile with `--strictNullChecks` but haven't verified.